st2client/st2client/commands/action.py

```diff
@@ -278,7 +278,44 @@ class ActionDisableCommand(resource.ContentPackResourceDisableCommand):
 
 
 class ActionDeleteCommand(resource.ContentPackResourceDeleteCommand):
-    pass
+    def __init__(self, resource, *args, **kwargs):
+        super(ActionDeleteCommand, self).__init__(resource, *args, **kwargs)
+
+        self.parser.add_argument(
+            "-f",
+            "--force",
+            action="store_true",
+            dest="force",
+            help="Auto yes flag to delete action files from disk.",
+        )
+
+    @add_auth_token_to_kwargs_from_cli
+    def run(self, args, **kwargs):
+        resource_id = getattr(args, self.pk_argument_name, None)
+        instance = self.get_resource(resource_id, **kwargs)
+        resource_delete_msg = 'Resource with id "{0}" has been successfully deleted from database and disk.'.format(
+            resource_id
+        )
+        if args.force:
+            self.manager.delete(instance, **kwargs)
+            print(resource_delete_msg)
+        else:
+            user_input = input(
+                "The resource files on disk will be deleted. Do you want to continue? (y/n): "
+            )
+            if user_input.lower() == "y" or user_input.lower() == "yes":
+                self.manager.delete(instance, **kwargs)
+                print(resource_delete_msg)
+            else:
+                print("Action is not deleted.")
+
+    def run_and_print(self, args, **kwargs):
+        resource_id = getattr(args, self.pk_argument_name, None)
```

Comment:
[This line](https://github.com/StackStorm/st2/blob/5587123977142467fba20eaa22edc3dbea248fa8/st2client/st2client/commands/resource.py#L127) will guarantee that `self.pk_argument_name` will at least be `None`.

And [this line](https://github.com/StackStorm/st2/blob/5587123977142467fba20eaa22edc3dbea248fa8/st2client/st2client/commands/resource.py#L758) will guarantee that it will be `"ref_or_id"`.

So you don't need the third argument to `getattr()` here.